### PR TITLE
sdk/ui: cleanup & improve types

### PR DESCRIPTION
**Pull request by kangarang**
this pr cleans up and improves types, mainly within the sdk package

### CODE REVIEW

File: packages/frontend/src/config/metadata.ts [hunk 1]
```diff
@@ -14,10 +14,11 @@ import wBtcLogo from 'src/assets/logos/wbtc.svg'
 import ethLogo from 'src/assets/logos/eth.svg'
 import maticLogo from 'src/assets/logos/matic.svg'
 import { network } from 'src/config'
+import { AllTokens, Slugs } from '@hop-protocol/sdk'
 
 type Metadata = {
   tokens: {
-    [key: string]: {
+    [key in AllTokens | string]: {
```
Comment (miguelmota) [inline packages/frontend/src/config/metadata.ts:21]: interesting, didn't know this was possible

File: packages/sdk/src/constants/constants.ts [hunk 2]
```diff
@@ -1,18 +1,74 @@
...
+
+export enum ChainNames {
+  Ethereum = 'Ethereum',
+  Optimism = 'Optimism',
+  Arbitrum = 'Arbitrum',
+  Polygon = 'Polygon',
+  Gnosis = 'Gnosis',
+}
+
+export enum ChainSlugs {
```
Comment (miguelmota) [inline packages/sdk/src/constants/constants.ts:24]: Shouldn't enum names be singular as best practice?

https://stackoverflow.com/a/1336012/1439168

File: packages/sdk/src/Base.ts [hunk 3]
```diff
@@ -3,13 +3,13 @@ import memoize from 'fast-memoize'
...
 import { Chain, Token as TokenModel } from './models'
-import { Chain as ChainEnum, MinPolygonGasPrice } from './constants'
+import { MinPolygonGasPrice, Slugs } from './constants'
 import { TChain, TProvider, TToken } from './types'
 import { config, metadata } from './config'
 import { getContractFactory, predeploys } from '@eth-optimism/contracts'
 import { parseEther, serializeTransaction } from 'ethers/lib/utils'
 
-export type ChainProviders = { [chain: string]: providers.Provider }
+export type ChainProviders = { [slug: string]: providers.Provider } | any
```
Comment (miguelmota) [inline packages/sdk/src/Base.ts:12]: why is `any` added here?

File: packages/sdk/src/constants/constants.ts [hunk 4]
```diff
@@ -1,18 +1,74 @@
...
+
+export enum HTokens {
+  hETH = 'hETH',
+  hMATIC = 'hMATIC',
+  hUSDC = 'hUSDC',
+  hUSDT = 'hUSDT',
+  hDAI = 'hDAI',
+}
+
+export type AllTokens = Tokens | WrappedTokens | HTokens
```
Comment (miguelmota) [inline packages/sdk/src/constants/constants.ts:70]: I find the `AllTokens` name a little confusing. A suggestion is to use `Token` as the type name and call the `Tokens` enum as `CanonicalToken`

File: packages/sdk/src/constants/constants.ts (hunk 2 above)
Comment (kangarang) [reply packages/sdk/src/constants/constants.ts]: i was debating this as well. the main reason for the plural naming was to minimize the number of duplicate name exports. this is now addressed in the latest commit

File: packages/sdk/src/Base.ts (hunk 3 above)
Comment (kangarang) [reply packages/sdk/src/Base.ts]: thanks! it's from when i was fiddling around.

File: packages/sdk/src/constants/constants.ts (hunk 4 above)
Comment (kangarang) [reply packages/sdk/src/constants/constants.ts]: how about:

```
AllTokens -> TokenSymbol
Tokens -> CanonicalToken
Networks -> NetworkSlug

ChainSlugs -> ChainSlug
Slugs -> Slug
ChainIds -> ChainId
WrappedTokens -> WrappedToken
HTokens -> HToken
```

Comment (miguelmota) [reply packages/sdk/src/constants/constants.ts]: sounds good to me 👍

**Review by miguelmota — APPROVED** [review]

**Pull request merged**